dissect/target/plugins/apps/webservers/apache.py

```diff
@@ -32,43 +128,76 @@ def infer_log_format(line: str) -> Optional[LogFormat]:
         vhost_combined = 'example.com:80 1.2.3.4 - - [19/Dec/2022:17:25:40 +0100] "GET / HTTP/1.1" 200 312 "-"
                           "Mozilla/5.0 (Windows NT 10.0; Win64; x64); AppleWebKit/537.36 (KHTML, like Gecko)
                           Chrome/108.0.0.0 Safari/537.36"'
-    """
 
-    first_part = line.split(" ")[0]
+    Two logformats encountered on Citrix Netscalers:
+        combined_resptime_with_citrix_hdrs  = '1.2.3.4 -> 192.168.4.20 - - [19/Dec/2022:17:25:12 +0100] [12311]
+                                               "GET / HTTP/1.1" 200 712 "-" "Mozilla/5.0
+                                               (Windows NT 10.0; Win64; x64); AppleWebKit/537.36 (KHTML, like Gecko)
+                                               Chrome/108.0.0.0 Safari/537.36" "Time: 11264 microsecs"'
+
+
+        combined_resptime                   = '127.0.0.1 - - [19/Dec/2022:17:25:12 +0100] [69195] "GET / HTTP/1.1" 200
+                                               18705 "-" "curl/7.78.0" "Time: 49835 microsecs"'
+
+    """
+    splitted_line = line.split(" ")
+    first_part = splitted_line[0]
+    second_part = splitted_line[1]
     if ":" in first_part and "." in first_part:
```

Comment:
Is it necessary to also check for the `.`? Cannot think of a scenario (other than a corrupt file)

Or maybe you mean `second_part` instead, to check for an IP-address?

Reply:
I think with vhosts, you have something like `example.com:80` as the first part of a line, which contains both a `:` and a `.`